src/lib/containers/widgets/facet-nav/FacetNav.tsx

```diff
@@ -0,0 +1,259 @@
+import * as React from 'react'
+import FacetNavPanel from './FacetNavPanel'
+import { applyChangesToValuesColumn } from '../query-filter/QueryFilter'
+import { QueryWrapperChildProps } from 'lib/containers/QueryWrapper'
+import {
+  FacetColumnResultValues,
+  FacetColumnResultValueCount,
+  FacetColumnRequest,
+  FacetColumnResult,
+} from 'lib/utils/synapseTypes'
+import { useState, useEffect } from 'react'
+import * as _ from 'lodash'
+
+export type FacetNavOwnProps = {
+  applyChanges: Function
+  loadingScreen?: React.FunctionComponent | JSX.Element
+  facetsToPlot?: string[] | undefined
+}
+
+type UiFacetState = {
+  name: string
+  isHidden: boolean
+  isExpanded: boolean
+  index?: number
+}
+
+type ExpandedFacet = {
+  facet: FacetColumnResult
+  index: number
+}
+
+type FacetNavProps = FacetNavOwnProps & QueryWrapperChildProps
+
+type ShowMoreState = 'MORE' | 'LESS' | 'NONE'
+
+const FacetNav: React.FunctionComponent<FacetNavProps> = ({
+  data,
+  getLastQueryRequest,
+  isLoadingNewData,
+  loadingScreen,
+  executeQueryRequest,
+  asyncJobStatus,
+  facetsToPlot,
+}: FacetNavProps): JSX.Element => {
+  const [facets, setFacets] = useState<FacetColumnResult[]>([])
+  const [facetUiStateArray, setFacetUiStateArray] = useState<UiFacetState[]>([])
+  const [expandedFacets, setExpandedFacets] = useState<ExpandedFacet[]>([])
+  const [isFirstTime, setIsFirstTime] = useState(true)
+
+  const request = getLastQueryRequest!()
+
+  useEffect(() => {
+    let result = data?.facets?.filter(
+      item =>
+        item.facetType === 'enumeration' &&
+        (!facetsToPlot?.length || facetsToPlot.indexOf(item.columnName) > -1),
+    )
+    if (!result) {
+      return
+    }
+    if (isFirstTime) {
+      setFacetUiStateArray(
```

Comment:
The problem is that initially there might not be data since it's an async call so while we do need to do it once we do need to do it after the data is present